pyproject.toml

```diff
@@ -28,7 +28,7 @@ classifiers = [
     "Topic :: Internet :: WWW/HTTP",
 ]
 dependencies = [
-    "anyio>=3.4.0,<5",
+    "anyio>=3.4.0,<4",
```

Comment:
Do I also remove this, then?

Reply:
I think that the only change you should make is to pin AnyIO to 3.* (and trio to 0.21) in your test requirements. The other changes don't hurt compatibility. But given that tests demonstrably do pass with AnyIO 4, users should be given the option of running Starlette against it.

Reply:
> But given that tests demonstrably do pass with AnyIO 4, users should be given the option of running Starlette against it.

I didn't get this... The tests are failing on main.

Reply:
Not the tests themselves, only the coverage part where 1 code line is missed, and looking at the code, it's not clear if that is a problem at all. That's the thing I asked you to investigate.